### ISSUE TITLE
🐛 BUG: Style tag ignored in components with slots

### ISSUE THREAD

**Issue by JoelHobson** [snowpackjs/astro#605]
### What package manager are you using?

npm

### What operating system are you using?

Windows 10

### Describe the Bug

If a component includes a `<slot />` tag, then the style tag in the component will be ignored if plain text or another Astro component is passed into the slot. If the component or text is wrapped by a div (and presumably any other plain HTML tag), then the styles are used. 

### Steps to Reproduce

1. `npm init astro` using template Starter Kit (Generic)
2. Add a `layouts` folder containing a Layout.astro file with this content:
```
---
---
<style>
    .content {
        background-color: red;
        width: 500px;
        height: 500px;
    }
</style>

<html>
    <head></head>
    <body>
        <div class="content">
            <slot />
        </div>
    </body>
</html>
```
3. Add a component called `Content.astro` to the components folder, with this content:
```
---
---
<div>Hello world!</div>
```
4. Replace the contents of `pages/index.astro` with the following:
```
---
import Layout from '../layouts/Layout.astro';
import Content from '../components/Content.astro';
---
<Layout>
    <Content />
</Layout>
```
5. Run the dev server and visit the page. You should see the text "Hello World" on a plain white background.
6. Change `pages/index.astro` so that `<Content />` is inside of a div tag.
7. Reload the page - there should now be a large red square behind the "Hello World"

Expected result: The styles in the layout are applied regardless of the content passed into the slot.

This issue also occurs if you replace `<Content />` in `index.astro` with plain text.

**Comment by natemoo-re** [comment]
This was a weird one to track down! The problem here was that we were injecting CSS (and HMR) into a page based on the existence of certain Elements, but your page in this case only has Components! That's why adding a `<div>` anywhere on the page solved this.

The fix actually didn't have anything to do with slots, but now we should inject CSS/HMR code into pages even if they consist of only components.

**Comment by JoelHobson** [comment]
Should this be resolved in 0.17.1? It still seems to be happening for me.
